File: medplum-java/medplum-model/src/main/java/com/fhir/Elementdefinition_typeVersioning.java
package com.fhir;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Locale;
import javax.annotation.Nonnull;

public final class Elementdefinition_typeVersioning {
  public static final Elementdefinition_typeVersioning INDEPENDENT = new Elementdefinition_typeVersioning(Value.INDEPENDENT, "INDEPENDENT");

  public static final Elementdefinition_typeVersioning EITHER = new Elementdefinition_typeVersioning(Value.EITHER, "EITHER");

  public static final Elementdefinition_typeVersioning SPECIFIC = new Elementdefinition_typeVersioning(Value.SPECIFIC, "SPECIFIC");

  private final Value value;

  private final String string;

  Elementdefinition_typeVersioning(Value value, String string) {
    this.value = value;
    this.string = string;
  }

  public Value getEnumValue() {
    return value;
  }

  @Override
  @JsonValue
  public String toString() {
    return this.string;
  }

  @Override
  public boolean equals(Object other) {
    return (this == other) 
      || (other instanceof Elementdefinition_typeVersioning && this.string.equals(((Elementdefinition_typeVersioning) other).string));
  }

  @Override
  public int hashCode() {
    return this.string.hashCode();
  }

  public <T> T accept(Visitor<T> visitor) {
    switch (value) {
      case INDEPENDENT:
        return visitor.visitINDEPENDENT();
      case EITHER:
        return visitor.visitEITHER();
      case SPECIFIC:
        return visitor.visitSPECIFIC();
      case UNKNOWN:
      default:
        return visitor.visitUnknown(string);
    }
  }

  @JsonCreator(
      mode = JsonCreator.Mode.DELEGATING
  )
  public static Elementdefinition_typeVersioning valueOf(@Nonnull String value) {
    String upperCasedValue = value.toUpperCase(Locale.ROOT);
    switch (upperCasedValue) {
      case "INDEPENDENT":
        return INDEPENDENT;
      case "EITHER":
        return EITHER;
      case "SPECIFIC":
        return SPECIFIC;
      default:
        return new Elementdefinition_typeVersioning(Value.UNKNOWN, upperCasedValue);
    }
  }

  public enum Value {
    EITHER,

    INDEPENDENT,

    SPECIFIC,

    UNKNOWN
  }

  public interface Visitor<T> {
    T visitEITHER();

    T visitINDEPENDENT();

    T visitSPECIFIC();

    T visitUnknown(String unknownType);
  }
}

File: medplum-typescript/src/model/fhir/Elementdefinition_typeVersioning.ts
export type Elementdefinition_typeVersioning =
    | Elementdefinition_typeVersioning.Either
    | Elementdefinition_typeVersioning.Independent
    | Elementdefinition_typeVersioning.Specific;

export const Elementdefinition_typeVersioning = {
    Either: "either" as Elementdefinition_typeVersioning.Either,
    Independent: "independent" as Elementdefinition_typeVersioning.Independent,
    Specific: "specific" as Elementdefinition_typeVersioning.Specific,

    _visit: <Result>(value: Elementdefinition_typeVersioning, visitor: Elementdefinition_typeVersioning._Visitor<Result>): Result => {
        switch (value) {
            case Elementdefinition_typeVersioning.Either: return visitor.either();
            case Elementdefinition_typeVersioning.Independent: return visitor.independent();
            case Elementdefinition_typeVersioning.Specific: return visitor.specific();
            default: return visitor._unknown();
        }
    },
};

export declare namespace Elementdefinition_typeVersioning {
    export type Either = "either" & {
        __Either: void,
    };
    export type Independent = "independent" & {
        __Independent: void,
    };
    export type Specific = "specific" & {
        __Specific: void,
    };

    export interface _Visitor<Result> {
        either: () => Result;
        independent: () => Result;
        specific: () => Result;
        _unknown: () => Result;
    }
}
